packages/frontend/features/swap-tokens/useSwap.ts

```diff
@@ -41,6 +41,7 @@ const useSwap = (thingToSwap, fromTokenStr, toTokenStr, amountToSwap) => {
         actionText: "Check",
         variant: "processing",
       });
+      setLoading(false);
```

Comment:
`setLoading(false)` should probably come after `tx.wait`

Reply:
I'm just mimicking the behavior describe below on line 61-63

Reply:
Ah, unfortunately it'll stop spinning before the transaction is even mined. I should probably fix that other part hen.